package/kedro_viz/launchers/jupyter.py

```diff
@@ -78,26 +80,48 @@ def _display_databricks_html(port: int):  # pragma: no cover
         print(f"Kedro-Viz is available at {url}")
 
 
-def run_viz(port: int = None, local_ns: Dict[str, Any] = None) -> None:
+def parse_args(args):  # pragma: no cover
+    """Parses the args string and returns a dictionary of arguments."""
+    parsed_args = shlex.split(args)
+    arg_dict = {
+        arg.lstrip("-").split("=")[0]: arg.split("=")[1] if "=" in arg else True
+        for arg in parsed_args
+    }
+    return arg_dict
+
+
+def run_viz(  # pylint: disable=too-many-locals
+    args: str = "", local_ns: Dict[str, Any] = None
+) -> None:
     """
-    Line magic function to start kedro viz. It calls a kedro viz in a process and displays it in
-    the Jupyter notebook environment.
+    Line magic function to start Kedro Viz with optional arguments.
 
     Args:
-        port: TCP port that viz will listen to. Defaults to 4141.
+        args: String of arguments to pass to Kedro Viz. If empty, defaults will be used.
         local_ns: Local namespace with local variables of the scope where the line magic
             is invoked. This argument must be in the signature, even though it is not
             used. This is because the Kedro IPython extension registers line magics with
             needs_local_scope.
             https://ipython.readthedocs.io/en/stable/config/custommagics.html
 
     """
-    port = port or DEFAULT_PORT  # Default argument doesn't work in Jupyter line magic.
-    host = _DATABRICKS_HOST if _is_databricks() else DEFAULT_HOST
-    port = _allocate_port(
-        host, start_at=int(port)
-    )  # Line magic provides string arguments by default, so we need to convert to int.
-
+    # Parse arguments
+    arg_dict = parse_args(args)
+
+    host = arg_dict.get("host", _DATABRICKS_HOST if _is_databricks() else DEFAULT_HOST)
+    port = int(arg_dict.get("port", DEFAULT_PORT))
+    load_file = arg_dict.get("load-file", None)
+    save_file = arg_dict.get("save-file", None)
+    pipeline = arg_dict.get("pipeline", None)
+    env = arg_dict.get("env", None)
```

Comment:
looks good overall. Are we not allowing shorthand in jupyter like (-a for autoreload, -e for env) ?

Reply:
We could but would need to add more parsing logic to `parse_args` and this becomes quite complicated especially given the number of arguments we have. At that point I think we should just use the click context somehow with the line magic but this is also very convoluted approach.

Given the complexities, I think adding these might be outside the scope of this PR.